### DISCUSSION TITLE
AppLaunchComplete" beacon is missing.

### DISCUSSION THREAD

**Discussion by awalt125** [rrirower/DP2SceneGraphWizard#51] [General]
I have a very simple live video channel and I am stuck at the AppLaunchComplete" beacon is missing.    I don't know were to put it I have tied every thing I have found on line and no luck   Can someone please help me

**Comment by rrirower** [comment]
Are you using this project to generate the Brightscript code?

**Reply by awalt125** [reply]
No I posted this in the wrong spot,  Can anyone point me in the correct direction? I am using a sample channel i found and edited I got every thing good except for the AppLaunchComplete" beacon is missing

**Reply by rrirower** [reply]
Do you need a custom layout? If not, and, you don't mind a Grid layout, it would be much easier for you to download this wizard tool and let it generate your code.  Otherwise, the call to that beacon should occur after the Main scene is set.

**Reply by awalt125** [reply]
Its kind of custom but I might Have to re do it,     Here is my homescene  code    can you tell me what to put in it and were to make the beacon work ?        Sub init()
  m.RowList = m.top.findNode("RowList")
    m.Title = m.top.findNode("Title")
    m.Description = m.top.findNode("Description")
   m.Poster = m.top.findNode("Poster")
    m.RowList.setFocus(true)
    m.LoadTask = CreateObject("roSGNode", "FeedParser") 
'Create XML parsing node task

    m.LoadTask.observeField("content", "rowListContentChanged")
    m.LoadTask.observeField("mediaIndex","indexloaded")
    m.LoadTask.control = "RUN"  'Run the task node

    m.InputTask=createObject("roSgNode","inputTask")
    m.InputTask.observefield("inputData","handleInputEvent")
    m.InputTask.control="RUN"

    m.RowList.observeField("rowItemFocused", "changeContent")

    m.Video = m.top.findNode("Video")
    m.Video.observeField("state", "onVideoStateChanged")
    m.VideoContent = createObject("roSGNode", "ContentNode")
    m.RowList.observeField("rowItemSelected", "playVideo")
End Sub

sub indexloaded(msg as Object)
    if type(msg) = "roSGNodeEvent" and msg.getField() = "mediaIndex"
        m.mediaIndex = msg.getData()
        ? "m.mediaIndex= "; m.mediaIndex
    end if
    handleDeepLink(m.global.deeplink)
    'get run time deeplink updates'
    'm.global.observeField("deeplink", handleRuntimeDeepLink)
end sub

Function handleDeepLink(deeplink as object)
  if validateDeepLink(deeplink)
    playVideo(m.mediaIndex[deeplink.id].url)
  else
    print "deeplink not validated"
  end if
end Function

sub handleInputEvent(msg)
    ? "in handleInputEvent()"
    if type(msg) = "roSGNodeEvent" and msg.getField() = "inputData"
        deeplink = msg.getData()
        if deeplink <> invalid
            handleDeepLink(deeplink)
        end if
    end if
end sub

function validateDeepLink(deeplink as Object) as Boolean
  mediatypes={movie:"movie",episode:"episode",season:"season",series:"series"}
  if deeplink <> Invalid
      ? "mediaType = "; deeplink.type
      ? "contentId = "; deeplink.id
      ? "content= "; m.mediaIndex[deeplink.id]
      if deeplink.type <> invalid then
        if mediatypes[deeplink.type]<> invalid
          if m.mediaIndex[deeplink.id] <> invalid
            if m.mediaIndex[deeplink.id].url <> invalid
              return true
            else
                print "invalid deep link url"
            end if
          else
            print "bad deep link contentId"
          end if
        else
          print "unknown media type"
        end if
      else
        print "deeplink.type string is invalid"
      end if
  end if
  return false
end function




Sub rowListContentChanged(msg as Object)
    if type(msg) = "roSGNodeEvent" and msg.getField() = "content"
        m.RowList.content = msg.getData()
    end if
end Sub

Sub changeContent()  'Changes info to be displayed on the overhang
    contentItem = m.RowList.content.getChild(m.RowList.rowItemFocused[0]).getChild(m.RowList.rowItemFocused[1])
    'contentItem is a variable that points to (rowItemFocused[0]) which is the row, and rowItemFocused[1] which is the item index in the row

    m.top.backgroundUri = contentItem.HDPOSTERURL  'Sets Scene background to the image of the focused item
    m.Poster.uri = contentItem.HDPOSTERURL  'Sets overhang image to the image of the focused item
    m.Title.text = contentItem.TITLE  'Sets overhang title to the title of the focused item
    m.Description.text = contentItem.DESCRIPTION  'Sets overhang description to the description of the focused item
End Sub

Sub playVideo(url = invalid)
    ? "url= "; url
    if type(url) = "roSGNodeEvent"   ' passed from observe callback'
        m.videoContent.url = m.RowList.content.getChild(m.RowList.rowItemFocused[0]).getChild(m.RowList.rowItemFocused[1]).URL
        'rowItemFocused[0] is the row and rowItemFocused[1] is the item index in the row
    else
        m.videoContent.url = url
    end if

    m.videoContent.streamFormat = "mp4"
    keepPlaying = false

    m.Video.content = m.videoContent
    m.Video.visible = "true"
    m.Video.control = "play"

    m.vector2danimation = m.top.FindNode("moveOverhangPanelUp")
    m.vector2danimation.repeat = false
    m.vector2danimation.control = "start"
End Sub

Function returnToUIPage()
    m.Video.visible = "false" 'Hide video
    m.Video.control = "stop"  'Stop video from playing

    m.vector2danimation = m.top.FindNode("moveOverhangPanelDown")
    m.vector2danimation.repeat = false
    m.vector2danimation.control = "start"
end Function

Function onVideoStateChanged(msg as Object)
  if type(msg) = "roSGNodeEvent" and msg.getField() = "state"
      if msg.getData() = "finished"
        returnToUIPage()
      end if
  end if
end Function

Function onKeyEvent(key as String, press as Boolean) as Boolean  'Maps back button to leave video
    if press
        if key = "back"  'If the back button is pressed
            if m.Video.visible
                returnToUIPage()
                return true
            else
                return false
            end if
       end if
    end if
end Function

**Reply by rrirower** [reply]
It typically goes in main.brs

**Comment by awalt125** [comment]
Here is my main.brs  

sub Main(args)
  screen = CreateObject("roSGScreen")  'Create Screen object
    m.port = CreateObject("roMessagePort")  'Create Message port
    screen.setMessagePort(m.port)  'Set message port to listen to screen

    scene = screen.CreateScene("HomeScene")  'Create HomeScene

     m.global = screen.getGlobalNode()
    ? "args= "; formatjson(args)      'pretty print AA'
    deeplink = getDeepLinks(args)
    ? "deeplink= "; deeplink
    m.global.addField("deeplink", "assocarray", false)
    m.global.deeplink = deeplink

    screen.show()

    while(true)  'Listens to see if screen is closed
        msg = wait(0, m.port)
        msgType = type(msg)
        if msgType = "roSGScreenEvent"
            if msg.isScreenClosed() then return
        end if
    end while
end sub

Function getDeepLinks(args) as Object
    deeplink = Invalid

    if args.contentid <> Invalid and args.mediaType <> Invalid
        deeplink = {
            id: args.contentId
            type: args.mediaType
        }
    end if

    return deeplink
end Function

**Reply by rrirower** [reply]
After createscene.

**Reply by awalt125** [reply]
I have tried and tried Some how I am not getting it right,  I keep getting main.brsSyntax Error when put the beacon in there,  I think i am getting the command wrong, can you tell me how it should read ?

**Reply by awalt125** [reply]
I tried putting the code     myScene.signalBeacon(“AppLaunchComplete”)   but keep getting Syntax Error no mater were i put it

**Reply by rrirower** [reply]
If you want a working example of that function call, download the wizard tool and let it generate your code. Then, unzip the resulting channel file and look at the code.

**Reply by rrirower** [reply]
@awalt125 Have you resolved this?  If so, please comment here.
